### ISSUE TITLE
Can't use etcd 2.2.4 anymore

### ISSUE THREAD

**Issue by hectorj** [Masterminds/glide#231]
You can check with this glide.yaml:
```yaml
package: testEtcdGlide
import:
- package: github.com/coreos/etcd
  version: 2.2.4
```

Here is what it gives with the new master (aaf314cbb79b0c6a1b419a099fffdbcf774b1875):

```
$ glide up
[INFO] Fetching updates for github.com/coreos/etcd.
[INFO] Detected semantic version. Setting version for github.com/coreos/etcd to v2.2.4.
[INFO] Resolving imports
[INFO] Found Godeps.json file.
[INFO] Setting version for google.golang.org/cloud to f20d6dcccb44ed49de45ae3703312cb46e627db1.
[INFO] Setting version for golang.org/x/net to 7dbad50ab5b31073856416cdcfeb2796d682f844.
[INFO] Setting version for github.com/golang/protobuf to 5677a0e3d5e89854c9974e1256839ee23f8233ca.
[INFO] Setting version for golang.org/x/oauth2 to 3046bc76d6dfd7d3707f6640f85e42d9c4050f50.
[ERROR] Dependency /tmp/gopath/src/testEtcdGlide/vendor/google.golang.org/api/container/v1beta1 failed to resolve: lstat /tmp/gopath/src/testEtcdGlide/vendor/google.golang.org/api/container/v1beta1: no such file or directory.
[ERROR] Failed to retrieve a list of dependencies: lstat /tmp/gopath/src/testEtcdGlide/vendor/google.golang.org/api/container/v1beta1: no such file or directory
```

But it works if I revert Glide to f8f06478c5db01a4e02bf04fa684e01aecb40451

**Comment by mattfarina** [comment]
There is a problem here. I'm trying to figure out the best way to handle it.

Under Glide 0.8.3 there are no errors. But `google.golang.org/cloud` is set to `f20d6dcccb44ed49de45ae3703312cb46e627db1` and it has an import of `google.golang.org/api/container/v1beta1`. Glide 0.8.3 checkout out `google.golang.org/api` as master where this package doesn't exist. So, it's not catching the problem even though it is present.

Glide on `master` catches this problem.

coreos, using Godep, vendors `google.golang.org/cloud` and it is listed in the `Godeps.json` file. But, `google.golang.org/api` is not vendored.

I think the fix is for Glide to walk the used dependency tree rather than all possibilities. Coreos must not be using the part of `google.golang.org/cloud` that calls to `google.golang.org/api/container/v1beta1`.

This may be solved by #220 or some of the other work we've talked about.

**Comment by hectorj** [comment]
I wasn't sure if it was a bug in the old version or the new one, thanks for the investigation. It should indeed be fixed when issue #166 is resolved.

Temporary fix in case someone has the exact same issue : add 
```yaml
ignore:
- google.golang.org/api
```
to your glide.yaml

Closing.
